service/actions/posts.js

```diff
@@ -1,17 +1,170 @@
+import {forceLogoutIfNecessary} from './helpers';
+import {Constants, PostsTypes} from 'service/constants';
+
+function dispatchReceivedPosts(posts, channelId, dispatch, getState) {
+    dispatch(batchActions([
+        {
+            type: PostsTypes.RECEIVED_POSTS,
+            data: posts,
+            channelId
+        },
+        {
+            type: PostsTypes.GET_POSTS_SUCCESS
+        }
+    ]), getState);
+}
+
+export function createPost(teamId, post) {
+    return async (dispatch, getState) => {
+        try {
+            dispatch({type: PostsTypes.CREATE_POST_REQUEST}, getState);
+            const posted = await Client.createPost(teamId, post);
+            dispatch(batchActions([
+                {
+                    type: PostsTypes.RECEIVED_POSTS,
+                    data: posted
+                },
+                {
+                    type: PostsTypes.CREATE_POST_SUCCESS
+                }
+            ]), getState);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch);
+            dispatch({type: PostsTypes.CREATE_POST_FAILURE, error}, getState);
+        }
+    };
 }
+
+export function editPost(teamId, post) {
+    return async (dispatch, getState) => {
+        try {
+            dispatch({type: PostsTypes.EDIT_POST_REQUEST}, getState);
+            const edited = await Client.editPost(teamId, post);
+            dispatch(batchActions([
+                {
+                    type: PostsTypes.RECEIVED_POSTS,
+                    data: edited
+                },
+                {
+                    type: PostsTypes.EDIT_POST_SUCCESS
+                }
+            ]), getState);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch);
+            dispatch({type: PostsTypes.EDIT_POST_FAILURE, error}, getState);
+        }
+    };
+}
+
+export function deletePost(teamId, post) {
+    return async (dispatch, getState) => {
+        try {
+            dispatch({type: PostsTypes.DELETE_POST_REQUEST}, getState);
+
+            await Client.deletePost(teamId, post.channel_id, post.id);
+            dispatch(batchActions([
+                {
+                    type: PostsTypes.POST_DELETED,
+                    data: post
+                },
+                {
+                    type: PostsTypes.DELETE_POST_SUCCESS
+                }
+            ]), getState);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch);
+            dispatch({type: PostsTypes.DELETE_POST_FAILURE, error}, getState);
+        }
+    };
+}
+
+export function removePost(post) {
+    return async (dispatch, getState) => {
+        dispatch({
+            type: PostsTypes.REMOVE_POST,
+            data: post
+        }, getState);
+    };
+}
+
+export function getPost(teamId, channelId, postId) {
+    return async (dispatch, getState) => {
+        try {
+            dispatch({type: PostsTypes.GET_POSTS_REQUEST}, getState);
```

Comment:
In case of getting posts I don't see why we need different requests types as `GET_POST_REQUEST`, `GET_POSTS_REQUEST`, `GET_POSTS_SINCE_REQUEST` and so on